mod/01_rating/set/abstract/02_sort_and_filter/filter.rb

```diff
@@ -0,0 +1,89 @@
+def key_type_id
+  raise Error, "key_type_id not defined"
+end
+
+def filter_by_key _key
+  raise Error, "filter_by_key not defined"
+end
+
+def filter cache
+  return unless cache
+  cache = all_without_values cache if Env.params["value"] == "none"
+  cache.select { |key, values| pass_filter? key, values }
+end
+
+def pass_filter? key, values
+  filter_by_key(key) &&
+    filter_by_value(values) &&
+    filter_by_year(values)
+end
+
+def filter_by_name metric
+  return true unless Env.params["name"].present?
+  metric.downcase.include?(Env.params["name"].downcase)
+end
+
+def filter_by_value values
+  unit =
+    if Env.params["value"].present?
+      Env.params["value"]
+    else
+      "exists"
+    end
+  return values.empty? if unit == "none"
+  return !values.empty? if unit == "exists"
+  within_recent? unit, values
+end
+
+def filter_by_year values
+  return true unless Env.params["year"].present?
+  values.any? { |v| v["year"] == Env.params["year"] }
```

Comment:
yeah, noticed that too but with all the other changes I decided not to touch that. I renamed the method that calls all these methods "pass_filter?" to make it a bit clearer